### PR TITLE
Fix external consistency of grouping layers so they don't jump to the top of the hierarchy

### PR DESCRIPTION
**Pull request by moOsama76**
Part of #1633 (fixes "Grouping" > "External Consistency").

### CODE REVIEW

**Comment by moOsama76** [comment]
Calculate insert_index when grouping multiple selected layers using Ctrl+G shortcut

File: editor/src/messages/portfolio/document/document_message_handler.rs [hunk 1]
```diff
@@ -1101,7 +1118,7 @@ impl DocumentMessageHandler {
 	/// When working with an insert index, deleting the layers may cause the insert index to point to a different location (if the layer being deleted was located before the insert index).
 	///
 	/// This function updates the insert index so that it points to the same place after the specified `layers` are deleted.
-	fn update_insert_index(&self, layers: &[LayerNodeIdentifier], parent: LayerNodeIdentifier, insert_index: isize) -> isize {
+	fn updatecalculated_insert_index(&self, layers: &[LayerNodeIdentifier], parent: LayerNodeIdentifier, insert_index: isize) -> isize {
```
Comment (0HyperCube) [inline editor/src/messages/portfolio/document/document_message_handler.rs:1121]: I'm not quite sure why this was renamed? If this was intentional could you please standardise the formatting with underscores between words? Thanks.

File: frontend/package.json [hunk 2]
```diff
@@ -56,7 +56,7 @@
 		"vite-multiple-assets": "1.2.6"
 	},
 	"optionalDependencies": {
-		"wasm-pack": "0.12.1"
+		"wasm-pack": "^0.0.0"
```
Comment (0HyperCube) [inline frontend/package.json:59]: I think this was probably accidental (web package management can be quite fragile) - would it be possible to revert this change as well as the changes to the package-lock.json files?

**Review by 0HyperCube — CHANGES_REQUESTED** [review]
Code looks good and it works well.

I think you might have accidentally committed some unrelated changes though.

**Comment by moOsama76** [comment]
Yes those are accidental, I will revert now

**Comment by moOsama76** [comment]
Mentioned wrong commits fixed

**Review by 0HyperCube — APPROVED** [review]
Thanks for cleaning that up.

**Comment by Keavon** [comment]
Please discard the removal of `package-lock.json` from this. And in the future, please be very careful to review every file that you're including in a commit so you know which specific lines you intend to commit (basically a sanity check self-code review before any commit).

**Comment by Keavon** [comment]
!build

**Comment by moOsama76** [comment]
Do you mean package-lock.json in main folder or frontend/package-lock.json,
I don't know how those changed at all

**Comment by Keavon** [comment]
`frontend/package-lock.json`, please do a self-code review here on GitHub and you'll see it. You can run `git restore --source origin/master frontend/package-lock.json` and commit it back.

**Comment by moOsama76** [comment]
Please check now

**Comment by Keavon** [comment]
Yes, that's corrected now. I will code review and QA this soon today.

**Comment by moOsama76** [comment]
Thank you

**Comment by Keavon** [comment]
Please also fix the CI errors (it looks like a formatting issue) so this can pass and build.

**Comment by moOsama76** [comment]
Sorry I don't know what are CI errors

**Comment by Keavon** [comment]
See the ❌ below the comment section? That's telling you that your code has something wrong with it. Either the tests failed, there's a formatting error, the code can't compile, or something else is wrong. In this case, you need to fix your code formatting. You should set up your editor so it automatically formats your code every time you save your file. This should come as part of our provided VS Code settings if you're using that editor, otherwise you'll need to figure that out yourself if you're using another editor.

**Comment by moOsama76** [comment]
Done 🫡

**Comment by Keavon** [comment]
!build

File: editor/src/messages/portfolio/document/document_message_handler.rs [hunk 3]
```diff
@@ -483,14 +496,14 @@ impl MessageHandler<DocumentMessage, DocumentInputs<'_>> for DocumentMessageHand
...
-					insert_index: -1,
+					insert_index: calculated_insert_index.unwrap_or(-1),
 				});
 				responses.add(PortfolioMessage::PasteIntoFolder {
 					clipboard: Clipboard::Internal,
 					parent: LayerNodeIdentifier::new_unchecked(folder_id),
 					insert_index: -1,
 				});
-
+				let folder_id = NodeId(generate_uuid()); // TODO: Either this is a bug, or its reasoning should be explained in a comment
```
Comment (Keavon) [inline editor/src/messages/portfolio/document/document_message_handler.rs:506]: This line looks like it was accidentally copied from above (line 490) and put here. But if it has a purpose, please explain it for me so we can discuss if it should be kept.

**Review by Keavon — CHANGES_REQUESTED** [review]
Please resolve the potentially extraneous line I commented on.

**Comment by Keavon** [comment]
I committed a change which improves the code to make it follow idiomatic Rust (avoiding mutation, avoiding needing to break from nested loops, etc.). Please look at the diff so you can learn from it.

**Comment by moOsama76** [comment]
Great I will

File: editor/src/messages/portfolio/document/document_message_handler.rs (hunk 3 above)
Comment (moOsama76) [reply editor/src/messages/portfolio/document/document_message_handler.rs:506]: Yes it's a mistake

**Review by Keavon — APPROVED** [review]

**Comment by Keavon** [comment]
!build
